gio/src/task.rs

```diff
@@ -60,10 +85,188 @@ impl Task {
+    pub fn is_return_on_cancel(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_get_return_on_cancel(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_had_error")]
+    pub fn had_error(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_had_error(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_return_error_if_cancelled")]
+    pub fn return_error_if_cancelled(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_return_error_if_cancelled(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_set_check_cancellable")]
+    pub fn set_check_cancellable(&self, check_cancellable: bool) {
+        unsafe {
+            ffi::g_task_set_check_cancellable(self.to_glib_none().0, check_cancellable.into_glib());
+        }
+    }
+
+    #[cfg(any(feature = "v2_60", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_60")))]
+    #[doc(alias = "g_task_set_name")]
+    pub fn set_name(&self, name: Option<&str>) {
+        unsafe {
+            ffi::g_task_set_name(self.to_glib_none().0, name.to_glib_none().0);
+        }
+    }
+
+    #[doc(alias = "g_task_set_return_on_cancel")]
+    pub fn set_return_on_cancel(&self, return_on_cancel: bool) -> bool {
+        unsafe {
+            from_glib(ffi::g_task_set_return_on_cancel(
+                self.to_glib_none().0,
+                return_on_cancel.into_glib(),
+            ))
+        }
+    }
+
+    #[doc(alias = "g_task_is_valid")]
+    pub fn is_valid(
+        result: &impl IsA<AsyncResult>,
+        source_object: Option<&impl IsA<glib::Object>>,
+    ) -> bool {
+        unsafe {
+            from_glib(ffi::g_task_is_valid(
+                result.as_ref().to_glib_none().0,
+                source_object.map(|p| p.as_ref()).to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "completed")]
+    pub fn connect_completed_notify<F: Fn(&Self) + 'static>(&self, f: F) -> SignalHandlerId {
+        unsafe extern "C" fn notify_completed_trampoline<V, F>(
+            this: *mut ffi::GTask,
+            _param_spec: glib::ffi::gpointer,
+            f: glib::ffi::gpointer,
+        ) where
+            V: ValueType,
+            F: Fn(&Task<V>) + 'static,
+        {
+            let f: &F = &*(f as *const F);
+            f(&from_glib_borrow(this))
+        }
+        unsafe {
+            let f: Box_<F> = Box_::new(f);
+            connect_raw(
+                self.as_ptr() as *mut _,
+                b"notify::completed\0".as_ptr() as *const _,
+                Some(transmute::<_, unsafe extern "C" fn()>(
+                    notify_completed_trampoline::<V, F> as *const (),
+                )),
+                Box_::into_raw(f),
+            )
+        }
+    }
+
+    #[doc(alias = "g_task_return_value")]
+    #[doc(alias = "g_task_return_boolean")]
+    #[doc(alias = "g_task_return_int")]
+    #[doc(alias = "g_task_return_pointer")]
+    #[doc(alias = "g_task_return_error")]
```

Comment:
I think we generate too many doc aliases...

Reply:
What problem does it cause?

Reply:
None, it's just unhelpful.

Reply:
Why? It only shows up if you search for exactly that string, i.e. exactly that C function name. And then you find what you're looking for.

Reply:
I guess you're right.

Reply:
If it was a fuzzy search like for the other things then I would agree with you